packages/graphile-utils/src/makeChangeNullabilityPlugin.ts

```diff
@@ -1,49 +1,128 @@
+        );
+      } else {
+        if (rest.length > 0) {
+          throw new Error(`Too many booleans passed`);
+        }
+        inner = nullableType;
+      }
+      return nullability ? inner : new GraphQLNonNull(inner);
+    }
+    function applyTypeNullability(
+      build: Build,
+      type: any,
+      shouldBeNullable: boolean | boolean[]
+    ): any {
+      const allNullabilities: boolean[] =
+        Array.isArray(shouldBeNullable)
+          ? shouldBeNullable
+          : [shouldBeNullable];
+      return allNullabilities.length > 0
+        ? applyNullability(build, type, allNullabilities)
+        : type;
+    }
     function changeNullability<
       Field extends GraphQLInputFieldConfig | GraphQLFieldConfig<any, any>
     >(field: Field, build: Build, context: Context<Field>): typeof field {
       const {
         Self,
         scope: { fieldName },
       } = context;
-      const typeRules = rules[Self.name];
+      const typeName = Self.name;
+      const typeRules = rules[typeName];
       if (!typeRules) {
         return field;
       }
       const shouldBeNullable = typeRules[fieldName];
       if (shouldBeNullable == null) {
         return field;
       }
-      const {
-        graphql: { getNullableType, GraphQLNonNull },
-      } = build;
-      const nullableType = getNullableType(field.type);
-      return {
-        ...field,
-        type: shouldBeNullable
-          ? nullableType
-          : nullableType === field.type
-          ? new GraphQLNonNull(field.type)
-          : field.type, // Optimisation if it's already non-null
-      };
+      const shouldApplyNullabilities: NullabilityOpts = {};
+      if (
+        typeof shouldBeNullable === "object" &&
+        !Array.isArray(shouldBeNullable)
+      ) {
+        shouldApplyNullabilities.type = shouldBeNullable.type;
+        shouldApplyNullabilities.args = shouldBeNullable.args;
+      } else {
+        shouldApplyNullabilities.type = shouldBeNullable;
+      }
+      const updatedField = field;
+      try {
+        if (shouldApplyNullabilities.type) {
+          updatedField.type = applyTypeNullability(
+            build,
+            field.type,
+            shouldApplyNullabilities.type
+          );
+        }
+        if (shouldApplyNullabilities.args) {
+          Object.entries(shouldApplyNullabilities.args).forEach(
+            ([argName, argShouldBeNullable]) => {
+              if (!("args" in field) || !field["args"]?.[argName]) {
+                console.warn(
+                  `warning: makeChangeNullabilityPlugin. For ${typeName} > ${fieldName}: can't apply nullability rule for non-existing arg ${argName}`
+                );
+                return;
+              }
+              const argType = field["args"][argName].type;
+              field["args"][argName].type = applyTypeNullability(
+                build,
+                argType,
+                argShouldBeNullable
+              );
```

Comment:
Let's make this more similar to the type handling code above

```suggestion
              arg.type = applyTypeNullability(
                build,
                arg.type,
                argShouldBeNullable
              );
```